src/components/SensorsMap/index.tsx

```diff
@@ -0,0 +1,315 @@
+import { FC, useState } from "react";
+import Link from "next/link";
+import { SensorsListRow } from "@components/SensorsListRow";
+import { ParsedSensorType } from "@lib/hooks/usePublicSensors";
+import { MarkerMap } from "@components/MarkerMap";
+import { Pagination, PaginationType } from "@components/Pagination";
+import { useRouter } from "next/router";
+import { SensorsListRowLoadingSkeleton } from "@components/SensorsListRowLoadingSkeleton";
+import { Alert } from "@components/Alert";
+import { useWindowSize } from "@lib/hooks/useWindowSize";
+import MapIcon from "../../../public/images/icons/16px/map.svg";
+import ListIcon from "../../../public/images/icons/16px/list.svg";
+import { Switch } from "@headlessui/react";
+import useClickOutside from "@lib/hooks/useClickOutside";
+import { SensorSymbol } from "@components/SensorSymbol";
+import { UserAvatar } from "@components/UserAvatar";
+import { CategoryIcon } from "@components/CategoryIcon";
+
+interface SensorsMapType {
+  sensors?: ParsedSensorType[];
+  paginationProps: Omit<
+    PaginationType,
+    "numberOfDisplayedPages" | "marginPagesDisplayed"
+  >;
+  sensorsAreLoading?: boolean;
+  error?: Error;
+}
+
+function isInViewport(el: HTMLLIElement): boolean {
+  const rect = el.getBoundingClientRect();
+  return (
+    rect.top >= 0 &&
+    rect.left >= 0 &&
+    rect.bottom <=
+      (window.innerHeight || document.documentElement.clientHeight) &&
+    rect.right <= (window.innerWidth || document.documentElement.clientWidth)
+  );
+}
+
+function scrollToTargetAdjusted(el: HTMLLIElement): void {
+  if (isInViewport(el)) return;
+  const paddingTop = 10;
+  const elementPosition = el.getBoundingClientRect().top;
+  const footerHeight =
+    document.querySelector("#__next > footer")?.getBoundingClientRect()
+      .height || 0;
+  const headerHeight =
+    document.querySelector("#__next > main > header")?.getBoundingClientRect()
+      .height || 0;
+  const maxScrollDist =
+    document.documentElement.scrollHeight -
+    window.outerHeight -
+    footerHeight +
+    headerHeight;
+  const offsetPosition =
+    elementPosition + window.pageYOffset - headerHeight - paddingTop;
+
+  window.scrollTo({
+    top: Math.min(offsetPosition, maxScrollDist),
+    behavior: "smooth",
```

Comment:
How about only adding `"smooth"` if the `prefers-reduced-motion` media feature is not set to `reduce`? This way we would honour a user's preferences regarding motion sickness.

This would also apply to the `flyTo` of the map then...

Reply:
Great idea! I will try to implement it!

Reply:
I did add a react hook called `useReducedMotion` and set the flyTo and scroll behavior accordingly